tests/unit/swagger/schemaDetails3.spec.ts

```diff
@@ -374,6 +374,27 @@ describe('Definition generation for OpenAPI 3.0.0', () => {
                 genericNestedArrayCharacter2: (propertyName, propertySchema) => {
                     expect(propertySchema.$ref).to.eq('#/components/schemas/GenericRequestTypeAliasModel2Array', `for property ${propertyName}.$ref`);
                 },
+                and: (propertyName, propertySchema) => {
+                    expect(propertySchema).to.deep.include({
+                        allOf: [
+                            { $ref: '#/components/schemas/TypeAliasModel1' },
+                            { $ref: '#/components/schemas/TypeAliasModel2' },
+                        ],
+                    });
+                },
+                referenceAnd: (propertyName, propertySchema) => {
+                    expect(propertySchema).to.deep.include({
+                        $ref: '#/components/schemas/TypeAliasModelCase1',
+                    });
+                },
+                or: (propertyName, propertySchema) => {
+                    expect(propertySchema).to.deep.include({
```

Comment:
Gotta test `additionalProperties` here.

Reply:
When would `additionalProperties` be set here / be true? :confused:

Reply:
I’m not sure. But it’s something we have to figure out. Any schema for an object will have it set to true or false. Things that aren’t objects will have it set to undefined. I think the tests above this should provide guidance.

Reply:
And if it’s undefined for this case, I’d still like to see an assertion that confirms that it’s supposed to be undefined and explains why.

I know that this seems too cautious, but it’s important so that we prevent future regressions.

Reply:
https://github.com/lukeautry/tsoa/pull/400/commits/62bac808fb811cab908398bf24e637a484c29ce5